ArduPlane/mode_autoland.cpp

```diff
@@ -88,38 +99,37 @@ bool ModeAutoLand::_enter()
     land_start_WP.change_alt_frame(Location::AltFrame::ABSOLUTE);
 
     /*
-      now create the initial target waypoint for the base leg. We
+      now create the initial target waypoint for the loitering to alt centered on base leg waypoint. We
       choose if we will do a right or left turn onto the landing based
       on where we are when we enter the landing mode
      */
     const float bearing_to_current_deg = wrap_180(degrees(land_start_WP.get_bearing(plane.current_loc)));
     const float bearing_err_deg = wrap_180(wrap_180(plane.takeoff_state.initial_direction.heading) - bearing_to_current_deg);
     const float bearing_offset_deg = bearing_err_deg > 0? -90 : 90;
-    const float base_leg_length = final_wp_dist * 0.333;
-    cmd[0].id = MAV_CMD_NAV_WAYPOINT;
+
+    // make the "base leg" the smaller of the waypoint loiter radius or 1/3 of the final WP distance
+    // this results in a neat landing no matter the loiter radius
+    const float base_leg_length = MIN(final_wp_dist * 0.333, fabsf(plane.aparm.loiter_radius));
```

Comment:
This hits the loiter radius varying with altitude issue. Loiters get bigger with altitude, what happens if the radius is larger than `final_wp_dist * 0.333`. 

You could use `plane.nav_controller->loiter_radius(fabsf(plane.loiter.radius))` to get the corrected radius, but that will be fore the altitude at which home is entered not the altitude at which the waypoint is.

Reply:
this assures that the base leg and loiter radius never gets bigger than 1/3 the final no matter what the WP_LOITER_RAD is...Tridge added this

Reply:
That is not the case because of the altitude of loiter radius is scaled in the L1 controller. For example do a normal RTL in SITL, you will find the distance to home is 30% larger than the RTL radius. With defaults `RTL_RADIUS` is zero and `WP_LOITER_RAD` is 80. So we would expect the loiter radius to be 80m However, if you try it you will find that it is ~96m. The extra 16m are due to the altitude, at sea level it would loiter at the expected altitude. 

In this case if your base leg is short enough the second waypoint will be inside the loiterer circle which will cause the breakout logic to do funny things. (that is one of the things the other heading lined up function handled better)

Reply:
up to Tridge...his code change